resources/js/features/user-game-activity/components/+root/UserGameActivityMainRoot.tsx

```diff
@@ -0,0 +1,57 @@
+import { type FC, memo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { BaseLabel } from '@/common/components/+vendor/BaseLabel';
+import { BaseSwitch } from '@/common/components/+vendor/BaseSwitch';
+import { GameHeading } from '@/common/components/GameHeading';
+import { UserBreadcrumbs } from '@/common/components/UserBreadcrumbs';
+import { usePageProps } from '@/common/hooks/usePageProps';
+
+import { UserGameActivityClientBreakdown } from '../UserGameActivityClientBreakdown';
+import { UserGameActivityTimeline } from '../UserGameActivityTimeline';
+import { UserGameSummarizedActivity } from '../UserGameSummarizedActivity';
+
+export const UserGameActivityMainRoot: FC = memo(() => {
+  const { game, player } = usePageProps<App.Platform.Data.PlayerGameActivityPageProps>();
+
+  const { t } = useTranslation();
+
+  const [isOnlyShowingAchievementSessions, setIsOnlyShowingAchievementSessions] = useState(false);
+
+  return (
+    <div>
+      <UserBreadcrumbs game={game} user={player} t_currentPageLabel={t('Activity')} />
+      <GameHeading game={game} wrapperClassName="!mb-1">
+        {t('Game Activity - {{user}}', { user: player.displayName })}
+      </GameHeading>
+
+      <div className="flex flex-col gap-5">
+        <div className="flex flex-col gap-2">
+          <div className="flex flex-col items-start justify-between gap-4 rounded-lg bg-embed p-3 sm:flex-row">
+            <UserGameActivityClientBreakdown />
+
+            <div className="flex items-center">
+              <BaseLabel
+                htmlFor="toggle-productive-sessions"
+                className="max-w-[220px] leading-4 text-neutral-300 light:text-neutral-900"
+              >
+                {t('Hide all player sessions where achievements were not earned')}
+              </BaseLabel>
+              <BaseSwitch
+                id="toggle-productive-sessions"
+                checked={isOnlyShowingAchievementSessions}
+                onCheckedChange={setIsOnlyShowingAchievementSessions}
+              />
+            </div>
+          </div>
```

Comment:
Possible idea for future feature: Hide all unlocks (or collapse sessions). That way just the duration of each session and its final rich presence would be shown to give a better overview of the player's journey.

Reply:
I think this is a good idea. We'll need to create a new base component to support the functionality, but I'll see what may be required beyond that to pull this off.

Another idea that occurred to me is finding a better utilization of this dead space under the playtime label:
![Screenshot 2025-01-01 at 2 59 56 PM](https://github.com/user-attachments/assets/f56b582b-7ef3-4f8b-b545-3149c3f80487)

Maybe this space could be used to summarize the session itself? ie: "4 achievements unlocked for 20 points"?